std/algorithm/sorting.d

```diff
@@ -167,18 +166,53 @@ bool isSorted(alias less = "a < b", Range)(Range r) if (isForwardRange!(Range))
 
         for (; !ahead.empty; ahead.popFront(), r.popFront(), ++i)
```

Comment:
For some reason `lockstep` doesn't work, because it isn't `pure` - will investigate this in another PR.